### ISSUE TITLE
invalid feature extracted when duplicate coords were given to 'features_at_coords' 

### ISSUE THREAD

**Issue by Yukichiii** [NVIDIA/MinkowskiEngine#217]
when I want to extract the feature at some given points, 
the original coords may be `[0.1, 0.2], [2.2, 2.6], [0.5, 0.7], [-0.6, -0.7]`
but after quantized, if the coordinate is duplicate, like `[0, 0], [2, 2], [0, 0], [-1, -1]` in this example, the feature will be invalid

```
    coords = torch.IntTensor([[0, 0], [0, 1], [0, 2], [2, 0], [2, 2]])
    feats = torch.FloatTensor([[1.1, 2.1, 3.1, 4.1, 5.1]]).t()
    X = SparseTensor(feats, coords)
    feats, valid_inds = X.features_at_coords(torch.IntTensor([[0, 0], [2, 2], [0, 0], [-1, -1]]))
    print(feats)
    print(valid_inds)
```

the result is 

```
tensor([[1.1000],
        [5.1000],
        [0.0000],
        [0.0000]])
tensor([0, 1])
```

How can I get valid feature of the third point which has duplicated coord after quantize?

I check the code of function `features_at_coords`,
```
        cm = self.coords_man

        self_key = self.coords_key
        query_key = cm.create_coords_key(query_coords)

        self_indices, query_indices = cm.get_kernel_map(
            self_key, query_key, kernel_size=1)
        query_feats = torch.zeros((len(query_coords), self._F.size(1)),
                                  dtype=self.dtype,
                                  device=self.device)
```
I find that you create new kernel map to find the correspondence between query_keys and self_keys, and the duplicated coords was missed after create_coords_key.
But even I set `allow_duplicate_coords=True`, I still can not get the mapping of all valid coords because the kernel only allow one-to-one mapping.

so how can i get the mapping of all duplicated query coords to voxels?

**Comment by Yukichiii** [comment]
Ok, I find a way to solve this problem.
```
    coords = torch.IntTensor([[0, 0], [0, 1], [0, 2], [2, 0], [2, 2]])
    feats = torch.FloatTensor([[1.1, 2.1, 3.1, 4.1, 5.1]]).t()
    X = SparseTensor(feats, coords)
    query_coords = torch.IntTensor([[0, 0], [2, 2], [0, 0], [-1, -1]])
    #feats, valid_inds = X.features_at_coords(query_coords)

    unique_map, inverse_map = ME.utils.sparse_quantize(query_coords, return_index=True, return_inverse=True)
    unique_queries = query_coords[unique_map]
    feats, valid_inds = X.features_at_coords(unique_queries)
    all_query_feats = feats[inverse_map]
    print(all_query_feats)
```

the result will be 
```
tensor([[1.1000],
        [5.1000],
        [1.1000],
        [0.0000]])
```

or if I want to get the mapping from X to query points
```
    coords = torch.IntTensor([[0, 0], [0, 1], [0, 2], [2, 0], [2, 2]])
    feats = torch.FloatTensor([[1.1, 2.1, 3.1, 4.1, 5.1]]).t()
    X = SparseTensor(feats, coords)
    query_coords = torch.IntTensor([[0, 0], [2, 2], [0, 0], [-1, -1]])
    #feats, valid_inds = X.features_at_coords(query_coords)

    unique_map, inverse_map = ME.utils.sparse_quantize(query_coords, return_index=True, return_inverse=True)
    unique_queries = query_coords[unique_map]

    self_key = X.coords_key
    cm = X.coords_man
    unique_query_key = cm.create_coords_key(unique_queries)

    self_indices, query_indices = cm.get_kernel_map(self_key, unique_query_key, kernel_size=1)
    mapping = torch.zeros(unique_queries.shape[0], dtype=torch.long) - 1
    mapping[query_indices[0]] = self_indices[0]
    mapping = mapping[inverse_map]
    print(mapping)
```

the result will be:
```
tensor([ 0,  4,  0, -1])
```

the invalid mapping is represented as -1.

Do you have more elegant or easy way to find the mapping?

**Comment by chrischoy** [comment]
Let me work on a solution on the v0.5 branch.

**Comment by chrischoy** [comment]
Fixed on 29edbac6e5adfc6d5f46165ae74e3db258d41d58
